apps/doji-frontend/src/pages/profile/[username]/index.tsx

```diff
@@ -0,0 +1,183 @@
+import toast from 'react-hot-toast'
+import { BsThreeDotsVertical } from 'react-icons/bs'
+import { FiMessageSquare, FiUserPlus } from 'react-icons/fi'
+import { useQuery } from 'react-query'
+
+import { getServerSideUser } from '@frontend/common/auth'
+import RatingPanel from '@frontend/components/Review/RatingPanel'
+import { httpClient } from '@frontend/services'
+import { fetchUserInformation } from '@frontend/services/fetcher'
+
+import { IMeResponseDTO, IProfileResponseDTO, IReportDTO } from '@libs/api'
+import { Achievement, CompactProfile, SessionCard } from '@libs/mui'
+
+interface ProfilePageProps {
+  user: IMeResponseDTO
+}
+
+const ProfilePage: React.FC<ProfilePageProps> = ({ user }) => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+  const openMenu = Boolean(anchorEl)
+
+  const handleMenuOpen = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const handleMenuClose = () => {
+    setAnchorEl(null)
+  }
+
+  const submitReport = async () => {
+    try {
+      await toast.promise(httpClient.post('report', { expertUsername: displayUser.username }), {
+        loading: 'Loading...',
+        success: 'Report this expert successful.',
+        error: 'You have already reported this expert',
+      })
+    } catch (error) {}
+  }
+
+  const addFriend = async () => {
+    // TODO wait for friend system api
+    //
+    // try {
+    //   await toast.promise(httpClient.post('report', { expertUsername: displayUser.username }), {
+    //     loading: 'Loading...',
+    //     success: 'Report this expert successful.',
+    //     error: 'You have already reported this expert',
+    //   })
+    // } catch (error) {}
+  }
+
+  const { data: userData } = useQuery('user', fetchUserInformation, { initialData: user })
+  const currentUser = userData
+
+  const router = useRouter()
+  const username = router.query.username as string
+
+  const { data, isError, isLoading, error } = useQuery<IProfileResponseDTO>(
+    ['/profile', username],
+    () => httpClient.get(`/profile/${username}`).then((res) => res.data),
+  )
+  const displayUser = data
+
+  if (isError) return <p>{`Error: ${error}`}</p>
+
+  if (isLoading) return <CircularProgress />
+
+  return (
+    <Stack spacing={1}>
+      <CompactProfile
+        username={displayUser.username}
+        displayName={displayUser.displayName}
+        profileUrl={displayUser.profilePictureURL}
+      />
+
+      {currentUser.username !== displayUser.username && (
+        <Stack spacing={4} direction={'row'}>
+          <Button fullWidth onClick={addFriend}>
+            <FiUserPlus style={{ marginRight: 8 }} /> Add Friend
+          </Button>
+          <Button
+            fullWidth
+            variant="outlined"
+            onClick={() => {
+              router.push('/chat')
+            }}
+          >
+            <FiMessageSquare style={{ marginRight: 8 }} /> Message
+          </Button>
```

Comment:
`Link` from next/link

```suggestion
  <Link href="/chat" passHref">
          <Button
            fullWidth
            variant="outlined"
          >
            <FiMessageSquare style={{ marginRight: 8 }} /> Message
          </Button>
     </Link>
```